tenets.md

```diff
@@ -0,0 +1,423 @@
+# AWS Encryption SDK Specification's tenets
+
+# Definitions
+
+**Conventions used in this document**
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD",
+"SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be
+interpreted as described in [RFC 2119](https://tools.ietf.org/html/rfc2119).
+
+# Background
+
+The AWS Encryption SDK (ESDK) specification serves
+as a large living design document.
+These tenets are the "motivating force"
+from which we derive the ESDK design.
+
+We're always open to new ones,
+if you can think of better ones
+and make a case for them.
+
+# Priorities
+
+When weighing up difficult trade-offs
+our ordered set of priorities are:
+
+1. Security
+1. Readability
+1. Ease of use
+1. Performance
+
+# Principles
+
+When making design decisions
+these principles influence our choices.
+New features SHOULD look to principles
+to support adoption.
+
+## The specification first
+
+## Summary
+
+We write changes in the specification
+before they are developed in an implementation.
+
+If we can not write down what a feature should do,
+we can't have any confidence that it will do what we want.
+
+Examples of recording
+the requirements before development
+include test driven development,
+and specification by example.
+
+## Motivation
+
+The purpose of having a single specification
+is to create alignment and accountability.
+We describe what a feature is,
+why such a feature is important,
+and how such a feature should be used.
+This is like Test Driven Development
+where we create the requirement
+before we create the implementation.
+
+We MUST communicate our requirements
+in order to implement them.
+No matter how careful we are,
+there will be some ambiguity.
+We resolve this ambiguity by
+including why we made specific choices.
+
+This drives us towards alignment
+to keep the implementations the same.
+As implementors find ambiguity
+we use the written specification,
+background, and intention
+to keep us accountable.
+
+This alignment goes beyond just
+the inter-operability of the message format.
+We document the ESDK as a single library.
+We want to be able to describe
+how the ESDK works independently of the implementation.
+Clever or unique features in an implementation
+are an opportunity for bugs.
+They also work against inter-operability,
+because a unique behavior in one implementation
+may be relied on by customers.
+If this behavior does not exist
+in other implementations
+then along this dimension
+these implementations are not inter-operability
+
+For customers this means
+as we design features
+they will be better understood
+and they develop over time they should trend
+towards a constant idea.
```

Comment:
What I'm trying to get at
is that if we understand a feature
_because we wrote it down_
then that feature will be better.

Or the converse, if we don't understand
what a feature is it will be harder
for customers to understand let alone use it.

Reply:
rephrased